src/interface/interface.cpp

```diff
@@ -446,34 +456,33 @@ void run_simulation(const int problem_id, const double external_time,
       }
     }
     time += time_step_size;
-    times.push_back(time);
-    states.push_back(std::move(state));
+    interface->times_[i] = time;
+    interface->states_[i] = state;
   }
   interface->state_ = state;
 
-  // Extract last cardiac cycle
-  if (interface->output_last_cycle_only_) {
-    states.erase(states.begin(), states.end() - interface->pts_per_cycle_);
-    times.erase(times.begin(), times.end() - interface->pts_per_cycle_);
-    T start_time = times[0];
-    for (auto& time : times) {
-      time -= start_time;
-    }
-  }
-
   // Write states to solution output vector
   if (output_solutions.size() != num_output_steps * system_size) {
     throw std::runtime_error("Solution vector size is wrong.");
   }
   int output_idx = 0;
   int soln_idx = 0;
-  for (int t = 0; t < num_output_steps; t++) {
-    auto state = states[t];
-    output_times[t] = times[t];
+  int start_idx = 0;
+  T start_time = 0.0;
+  if (interface->output_last_cycle_only_) {  // NOT TESTED
+    start_idx = interface->num_time_steps_ - interface->pts_per_cycle_;
+    start_time = interface->times_[start_idx];
+  }
```

Comment:
Does this replace the functionality that was deleted above? If yes, maybe add a quick test that has `output_last_cycle_only=true` (and remove the `NOT TESTED` comment :smile:). If not, you can comment it and throw a "not implemented"-error if `output_last_cycle_only==true`.

Reply:
Added error message.